### ISSUE TITLE
Remove reference to Hydration in the docs

### ISSUE THREAD

**Issue by jgfet** [angular/angular#18458]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

https://angular.io/api/core/ChangeDetectionStrategy
Mentions word 'hydration' twice down at footer within the context of:
OnPush / Default:
https://angular.io/guide/glossary#H
No terms beginning with H

## Expected behavior

An explanation of Hydration , or a link to a blog post, video etc..
There is this info I was able to find here: (ProtoView???) https://github.com/m00s/angular2features
Quote:
"Views transition through a particular set of states:

View is created from the ProtoView.
View can be attached to an existing ViewContainerRef.
Upon attaching View to the ViewContainerRef the View needs to be hydrated. The hydration process involves instantiating all of the Directives associated with the current View.
At this point the view is ready and renderable. Multiple changes can be delivered to the Directives from the ChangeDetection.
At some point the View can be removed. At this point all of the directives are destroyed during the dehydration process and the view becomes inactive.
The View has to wait until it is detached from the DOM. The delay in detaching could be caused because an animation is animating the view away.
After the View is detached from the DOM it is ready to be reused. The view reuse allows the application to be faster in subsequent renderings."

===

A ProtoView is a compiled DOM View which is efficient at creating Views.
View contains a context object. The context represents the object instance against which all expressions are evaluated.
View contains a ChangeDetector for looking for detecting changes to the model.
View contains ElementInjector for creating Directives.

Maybe this can be reworded more simply in layman's terms?
==
## What is the motivation / use case for changing the behavior?

To inform users to the meaning of Angular jargon

## Environment

<pre><code>
Angular version: X.Y.Z

It's on website
Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by tbosch** [comment]
We used to have a way of reusing old instances of change detectors, and "hydration" applied to this. However, we don't do this anymore, so we should update the docs to not refer to this word anymore. In this case, it would just be "when the change detector is created". Updated on master already.

**Comment by tbosch** [comment]
Closed via 02394d2d8021c26c4ab80d89efcbba436120d96f.
